File: src/main/java/org/xmolecules/ide/intellij/All.java
package org.xmolecules.ide.intellij;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class All {
    public static final List<ConceptImplementation> ALL = concat(JMolecules.ALL , Spring.ALL, JPA.ALL);

    private static <T> List<T> concat(List<T> ... list) {
        var size = 0;
        for( var l : list) {
            size += l.size();
        }
        var buffer = new ArrayList<T>(size);
        for( var l : list) {
            buffer.addAll(l);
        }
        return Collections.unmodifiableList(buffer);
    }
}

File: src/main/kotlin/nexos/intellij/ddd/all.kt
package nexos.intellij.ddd

import nexos.intellij.ddd.jddd.jDDD
import nexos.intellij.ddd.jmolecules.jMolecules
import nexos.intellij.ddd.jpa.JPA

val all by lazy { jDDD.all + jMolecules.all + JPA.all}